From 38937cab27a3cb08360290acfd6c8d5cc92f9eb0 Mon Sep 17 00:00:00 2001
From: Geoff Lang <geofflang@chromium.org>
Date: Wed, 22 May 2024 14:20:31 +0000
Subject: [PATCH] Only call eglDestroySync on valid sync objects.

EGL_NO_SYNC is not a valid sync and generates errors or causes ANGLE to
crash when validation is disabled.

When GLFenceEGL fails to initialize, sync_ is left as EGL_NO_SYNC.

Bug: chromium:341595874, chromium:341388853, angleproject:8434
Change-Id: Ie00cd52d15c2549d2ad215ccc180af61723adfef
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5554177
Reviewed-by: Peng Huang <penghuang@chromium.org>
Commit-Queue: Geoff Lang <geofflang@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1304414}
---

--- a/ui/gl/gl_fence_egl.cc
+++ b/ui/gl/gl_fence_egl.cc
@@ -129,7 +129,9 @@
 }
 
 GLFenceEGL::~GLFenceEGL() {
-  eglDestroySyncKHR(display_, sync_);
+  if (sync_ != EGL_NO_SYNC) {
+    eglDestroySyncKHR(display_, sync_);
+  }
 }
 
 }  // namespace gl
